### PR TITLE
Fixed problems with server interfaces

### PR DESCRIPTION
**Pull request by jjkola**
Includes fixes to following components: i-hunchentoot, i-woo and i-wookie.
With these changes i-woo is fully working as far as I can see.
i-wookie needs more changes to work but at least it will start and the easy problems are corrected.
i-hunchentoot now correctly handles post request with non-form request

### CODE REVIEW

File: i-woo/i-woo.lisp [hunk 1]
```diff
@@ -182,12 +182,17 @@
...
-              (destructuring-bind (key val)
-                  (uiop:split-string pair :separator '(#\=))
+              (let* ((search-eq (position #\= pair))
+                     (key (subseq pair 0 search-eq))
+                     (val (subseq pair (1+ search-eq))))
                 (setf (gethash (string-trim '(#\space) key) cookies)
                       (string-trim '(#\space) val)))
             finally (return cookies)))))
 
+(defun strip-set-cookie (full-cookie)
```
Comment (Shinmera) [inline i-woo/i-woo.lisp]: A better name would be `extract header value`

File: i-woo/i-woo.lisp [hunk 2]
```diff
@@ -182,12 +182,17 @@
     (when cookies-header
       (loop with cookies = (make-hash-table :test 'equalp)
             for pair in (uiop:split-string cookies-header :separator '(#\;)) do
-              (destructuring-bind (key val)
-                  (uiop:split-string pair :separator '(#\=))
+              (let* ((search-eq (position #\= pair))
```
Comment (Shinmera) [inline i-woo/i-woo.lisp]: A better name would be `=-pos`

File: i-wookie/server.lisp [hunk 3]
```diff
@@ -100,7 +100,7 @@
       (etypecase (data response)
         (pathname
          (let ((buffer (make-array 1024 :element-type '(unsigned-byte 8)))
-               (output-stream (wookie:start-response wk-response :status (return-code response))))
+               (output-stream (wookie:start-response wk-response :status (return-code response) :headers (list :content-type (trivial-mimes:mime (data response))))))
```
Comment (Shinmera) [inline i-wookie/server.lisp]: The response headers should be included.

Also, proper file serving will cause the content-type to be set in the response object, like how you're using it below. Explicitly using trivial-mimes should be unnecessary.

File: i-wookie/server.lisp [hunk 4]
```diff
@@ -109,7 +109,7 @@
            (wookie:finish-response wk-response)))
         
         ((or string (array (unsigned-byte 8)))
-         (wookie:send-response wk-response :status (return-code response) :body (data response)))
+         (wookie:send-response wk-response :status (return-code response) :body (data response) :headers (list :content-type (or (content-type response) "text/plain"))))
```
Comment (Shinmera) [inline i-wookie/server.lisp]: Same as previous comment.

File: i-woo/i-woo.lisp (hunk 1 above)
Comment (jjkola) [reply i-woo/i-woo.lisp]: Done

File: i-woo/i-woo.lisp (hunk 2 above)
Comment (jjkola) [reply i-woo/i-woo.lisp]: Done

File: i-wookie/server.lisp (hunk 3 above)
Comment (jjkola) [reply i-wookie/server.lisp]: There is no need to include all the headers. Headers parameter are only used to add extra headers on top of response headers. But you are right there is no need to use trivial-mimes. Just use content-type from response.

File: i-wookie/server.lisp (hunk 4 above)
Comment (jjkola) [reply i-wookie/server.lisp]: That header parameter is only extra headers on top of response headers thus no need to include response headers. Removed unnecessary or-method and used response content-type directly.

**Comment by jjkola** [comment]
That wookie header parameter is only extra headers on top of response headers thus no need to include response headers.

**Pull request merged**